experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/ApplicationMaster.scala

```diff
@@ -0,0 +1,193 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gearpump.experiments.yarn
+
+import org.apache.gearpump.cluster.main.ArgumentsParser
+import org.apache.gearpump.cluster.main.CLIOption
+import org.apache.hadoop.yarn.api.ApplicationConstants
+import org.apache.hadoop.yarn.conf.YarnConfiguration
+import org.apache.hadoop.yarn.client.api.AMRMClient
+import org.apache.hadoop.yarn.client.api.NMClient
+import org.apache.hadoop.yarn.util.Records
+import org.apache.hadoop.yarn.api.records.Priority
+import org.apache.hadoop.yarn.api.records.Resource
+import org.apache.hadoop.yarn.client.api.async.AMRMClientAsync
+import org.apache.hadoop.yarn.api.protocolrecords.RegisterApplicationMasterResponse
+import org.apache.hadoop.yarn.client.api.AMRMClient.ContainerRequest
+import org.slf4j.Logger
+import org.apache.gearpump.util.LogUtil
+import org.apache.hadoop.yarn.client.api.async.impl.NMClientAsyncImpl
+import org.apache.hadoop.yarn.client.api.async.NMClientAsync
+import org.apache.hadoop.yarn.api.records.ContainerId
+import java.nio.ByteBuffer
+import org.apache.hadoop.yarn.api.records.ContainerStatus
+import java.util.concurrent.atomic.AtomicInteger
+import org.apache.hadoop.yarn.api.records.Container
+import scala.collection.JavaConverters._
+import org.apache.hadoop.yarn.api.records.NodeReport
+import org.apache.hadoop.yarn.api.records.ContainerExitStatus
+import org.apache.hadoop.yarn.api.records.ContainerLaunchContext
+import org.apache.hadoop.yarn.api.records.LocalResource
+import org.apache.hadoop.yarn.api.records.FinalApplicationStatus
+
+/**
+ * Yarn ApplicationMaster. 
+ */
+class ApplicationMaster(amArgs: ApplicationMasterArguments, yarnConf : YarnConfiguration) {
```

Comment:
Maybe rename this to YarnAppMaster so that we don't confuse this with Gearpupump AppMaster